### PR TITLE
#926 Use Webjars for JQueryUI

### LINKED ISSUE #926: Use webjars for JQuery UI
JQuery 1.13.2 is being bundled in:

- `wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui/src/main/resources/org/wicketstuff/jquery/ui/resource/jquery-ui.js`
- `wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui/src/main/resources/org/wicketstuff/jquery/ui/resource/jquery-ui.min.js`

### PR DESCRIPTION
**Pull request by mcr-paulanand**
Resolves #926

### CODE REVIEW

File: wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui/src/main/java/org/wicketstuff/jquery/ui/resource/JQueryUIResourceReference.java [hunk 1]
```diff
@@ -45,6 +51,25 @@ public static JQueryUIResourceReference get()
 	 */
 	private JQueryUIResourceReference()
 	{
-		super(JQueryUIResourceReference.class, "jquery-ui.js"); // v1.11.1
+		super("jquery-ui/current/jquery-ui.min.js");
```
Comment (solomax) [inline wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui/src/main/java/org/wicketstuff/jquery/ui/resource/JQueryUIResourceReference.java:54]: I believe `.min` should be removed here ...

**Review by solomax — CHANGES_REQUESTED** [review]
I would also 

 - use `*.css` from webjars for this `base` theme wicket-jquery-ui-parent/wicket-jquery-ui-themes/theme-base/src/main/java/com/googlecode/wicket/jquery/ui/theme/Initializer.java
 - `WicketWebjars.install(this);` should be added to `wicket-jquery-ui-parent/wicket-jquery-ui-samples/src/main/java/com/googlecode/wicket/jquery/ui/samples/SampleApplication.java`, otherwise examples will not work ....

File: wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui-themes/wicketstuff-jquery-ui-theme-base/src/main/java/org/wicketstuff/jquery/ui/theme/Initializer.java [hunk 2]
```diff
@@ -33,7 +34,7 @@ public void init(Application application)
 	{
 		application.getMarkupSettings().setStripWicketTags(true);
 
-		JQueryUILibrarySettings.get().setStyleSheetReference(new CssResourceReference(Initializer.class, "jquery-ui.css"));
+		JQueryUILibrarySettings.get().setStyleSheetReference(new WebjarsCssResourceReference("jquery-ui/current/jquery-ui.css"));
```
Comment (solomax) [inline wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui-themes/wicketstuff-jquery-ui-theme-base/src/main/java/org/wicketstuff/jquery/ui/theme/Initializer.java:37]: @mcr-paulanand correspondent `*.css` files need to be removed from sources :)

**Review by solomax — APPROVED** [review]

**Comment by solomax** [comment]
Hello @martin-g,

It seems this PR breaks samples

Webjars are being used from `Initializer` from now on

so we got 
```
Caused by: java.lang.IllegalStateException: you have to call WicketWebjars.install() before you can use an IWebjarsResourceReference or any other component.
    at de.agilecoders.wicket.webjars.WicketWebjars.settings (WicketWebjars.java:136)
    at de.agilecoders.wicket.webjars.util.WebjarsVersion$Holder.<clinit> (WebjarsVersion.java:29)
    at de.agilecoders.wicket.webjars.util.WebjarsVersion.useRecent (WebjarsVersion.java:45)
    at de.agilecoders.wicket.webjars.request.resource.WebjarsJavaScriptResourceReference.<init> (WebjarsJavaScriptResourceReference.java:29)
    at org.wicketstuff.jquery.ui.resource.JQueryUIResourceReference.<init> (JQueryUIResourceReference.java:54)
    at org.wicketstuff.jquery.ui.resource.JQueryUIResourceReference.<clinit> (JQueryUIResourceReference.java:37)
    at org.wicketstuff.jquery.ui.settings.JQueryUILibrarySettings.<init> (JQueryUILibrarySettings.java:49)
    at org.wicketstuff.jquery.ui.settings.JQueryUILibrarySettings.get (JQueryUILibrarySettings.java:61)
    at org.wicketstuff.jquery.ui.theme.Initializer.init (Initializer.java:37)
    at org.apache.wicket.Application.initInitializers (Application.java:561)
    at org.apache.wicket.Application.initApplication (Application.java:767)
    at org.apache.wicket.protocol.http.WicketFilter.init (WicketFilter.java:441)
```

due to `WicketWebjars.install(this);` is being called in the `init()` i.e. too late :(

What can be done here?

**Comment by martin-g** [comment]
As error message suggests we need to add 'WicketWebjars.install()' to the respective WebApplication#init()

**Comment by martin-g** [comment]
Oh, now I see the text after the stacktrace. 
I think it should be OK to call it in an initializer too.

**Comment by solomax** [comment]
Works as expected!

Thanks a million @martin-g :)

**Comment by reckart** [comment]
> due to WicketWebjars.install(this); is being called in the init() i.e. too late :(

@solomax so how did you solve this?

**Comment by solomax** [comment]
> @solomax so how did you solve this?

https://github.com/wicketstuff/core/blob/master/wicketstuff-jquery-ui-parent/wicketstuff-jquery-ui-themes/wicketstuff-jquery-ui-theme-base/src/main/java/org/wicketstuff/jquery/ui/theme/Initializer.java#L36

:))

**Comment by reckart** [comment]
I think this change needs to be made in the initializer of **all** of the themes separately.

**Comment by solomax** [comment]
> I think this change needs to be made in the initializer of **all** of the themes separately.

I also thought so

But I believe right now it's not necessary due to the fact the only theme uses webjar so far :)

**Comment by mcr-paulanand** [comment]
> > I think this change needs to be made in the initializer of **all** of the themes separately.
> 
> I also thought so
> 
> But I believe right now it's not necessary due to the fact the only theme uses webjar so far :)

I have created a PR #934 to resolve this

**Pull request merged**